filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceCompressor.java

```diff
@@ -114,9 +118,17 @@ private static void writeFileToTar(ArchiveOutputStream taos, File baseDir, File
     }
 
     private OutputStream compressedOutputStream(File outputFile) throws IOException {
+        log.log(Level.FINE, () -> "Compressing with type " + type + " and compression type " + compressionType);
         switch (type) {
             case compressed:
-                return new GZIPOutputStream(new FileOutputStream(outputFile));
+                switch (compressionType) {
+                    case gzip:
+                        return new GZIPOutputStream(new FileOutputStream(outputFile));
+                    case lz4:
+                        return new LZ4BlockOutputStream(new FileOutputStream(outputFile));
+                    default:
```

Comment:
What about zstandard ?

Reply:
output and input streams for zstandard not available yet, work in progress